### PR TITLE
Initial microblx interface

### PR DESCRIPTION
**Pull request by scymtym**
Initial mircoblx block for the bullet-based simulation.

### CODE REVIEW

**Review by svenschneider — COMMENTED** [review]
Looks mostly fine, I would just suggest the following changes:
- The `const` modifiers should remain in place since the `sim2b_bullet_nbx` represents the view from _inside_ the nanblock. Instead I suggest to include the data blocks into the `sim2b_bullet_ubx_private` structure (see https://github.com/rosym-project/robif2b/blob/master/src/ubx/2dof/ubx_2dof.c for an example). That also should take care of the memory leak from allocating the buffers.
- Add a LUA file describing the microBlock.
- A few minor points in-line.

File: src/ubx/bullet.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  if (nr_joints_len != 1) {
+    ubx_err(b, "sim2b_bullet: nr_joints must be supplied");
+    goto out;
+  }
+  nblock->nr_joints = *nr_joints_temp;
+
+  const double *gravity_temp;
+  int gravity_len = cfg_getptr_double(b, "gravity", &gravity_temp);
+  if (gravity_len == 3) {
+    memcpy(nblock->grav, gravity_temp, gravity_len*sizeof(double));
```
Comment (svenschneider) [inline src/ubx/bullet.c]: Would you mind adding spaces around the multiplication operator (also in the lines below)?
`gravity_len * sizeof(double)`

File: examples/bullet.usc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   -- connect blocks
+   connections = {
+   },
+
+   -- configure blocks
+   configurations = {
+      { name="sim", config = { nr_joints=7,
+                               gravity={0,0,-9.81},
+                               jnt_pos_init={1,2,3,4,5,6,7},
+                               urdf="/tmp/2b-makefile/vericomp-application@master/kortex_description/arms/gen3/7dof/urdf/gen3.urdf" } },
```
Comment (svenschneider) [inline examples/bullet.usc]: Could we have the `2dof.urdf` robot from the stand-alone example also here? Then it was possible to directly run this example script.

File: src/ubx/bullet.c (hunk 1 above)
Comment (scymtym) [reply src/ubx/bullet.c]: Done (not yet pushed).

File: examples/bullet.usc (hunk 2 above)
Comment (scymtym) [reply examples/bullet.usc]: I can change the degrees of freedom and the urdf file, but what should the directory layout be like?

Comment (svenschneider) [reply examples/bullet.usc]: We could move the `2dof.urdf` example file to the repository root (e.g. `/sim2b/misc/2dof.urdf`) and then copy it to the build root like in the stand-alone example (cf. https://github.com/rosym-project/sim2b/blob/master/src/example/CMakeLists.txt#L5). Do you think that's reasonable?

I also just noticed that the `examples` folder is in the repository root, too. For consistency, I would prefer to have all framework-related items in one place (see https://github.com/rosym-project/robif2b/tree/master/src/ubx/2dof for an example).

**Comment by scymtym** [comment]
> That also should take care of the memory leak from allocating the buffers.

The buffer sizes are not statically known, though.

**Pull request merged**